cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/OAuthHandler.java

```diff
@@ -208,21 +209,44 @@ public void putOAuthCredential(HttpServiceRequest request, HttpServiceResponder
             HttpURLConnection.HTTP_INTERNAL_ERROR, "Failed to parse JSON: " + e.getMessage(), e);
       }
 
-      if (refreshTokenResponse.getRefreshToken() == null || refreshTokenResponse.getRefreshToken().isEmpty()) {
+      boolean hasRefreshToken = refreshTokenResponse.getRefreshToken() != null
+          && !refreshTokenResponse.getRefreshToken().isEmpty();
+      boolean hasAccessToken = refreshTokenResponse.getAccessToken() != null
+          && !refreshTokenResponse.getAccessToken().isEmpty();
+
+      if (!hasAccessToken && !hasRefreshToken) {
         throw new OAuthServiceException(
-            HttpURLConnection.HTTP_INTERNAL_ERROR, "Refresh token response body did not contain refresh token");
+            HttpURLConnection.HTTP_INTERNAL_ERROR,
```

Comment:
Although unrelated to this PR but this shouldn't be `5xx`, please change it to `HTTP_BAD_REQUEST`

This will help plugin validations to not retry on `4xx`.

Reply:
Done